dipy/io/stateful_tractogram.py

```diff
@@ -161,7 +180,7 @@ def streamlines(self, streamlines):
         self._tractogram._streamlines = Streamlines(streamlines)
         self.data_per_point = self.data_per_point
         self.data_per_streamline = self.data_per_streamline
-        logging.warning('Streamlines has been modified')
+        logging.warning('Streamlines has been modified, caution')
```

Comment:
Well, it could be that the use shuffled them, resampled them, smoothed them or even took a subset (if there was not data_per_point or data_per_streamlines)

There is no way of knowing what changed exactly, that's why it is a "dangerous" operation and creating a new statefulTractogram is recommanded.

Any suggestion ? Personally I don't know what could be display to the user